### DISCUSSION TITLE
Find a library in /usr/local/lib?

### DISCUSSION THREAD

**Discussion by rjzak** [rui314/mold#1202] [Q&A] [answered]
I've got a project looking for zlib-ng, which is installed as:

```shell
❯ ls -la /usr/local/lib/libz-ng* 
-rw-r--r-- 1 root root 249590 Feb 12 10:11 /usr/local/lib/libz-ng.a
lrwxrwxrwx 1 root root     16 Feb 12 10:11 /usr/local/lib/libz-ng.so -> libz-ng.so.2.1.6
lrwxrwxrwx 1 root root     16 Feb 12 10:11 /usr/local/lib/libz-ng.so.2 -> libz-ng.so.2.1.6
-rwxr-xr-x 1 root root 214904 Feb 12 10:11 /usr/local/lib/libz-ng.so.2.1.6
```

Without Mold, the gcc linker finds it, but Mold does not. Does mold use the paths in `/etc/ld.so.conf`? This project also uses CMake, for which I could have probably done a better job with the `CMakeLists.txt` file :)

**Comment by rui314** [comment]
Maybe your `CMakeLists.txt` does not use `find_library` to find `libz-ng`? I believe the macro automatically adds `-L/usr/local/lib` if it found the specified library in `/usr/local/lib`.

**Reply by rjzak** [reply]
Seems not to, at least for me. I suppose I don't know why gcc/ld finds it and mold didn't, but it seems my CMake file wasn't being explicit with the library path.

**Reply by rui314** [reply, marked as answer]
GNU ld hardcodes some library paths, while mold and lld do not, facilitating cross-compilation. In general, it is not a good idea to rely on implicit library paths. You may want to fix your CMake file.
